x-pack/plugins/security_solution/common/risk_engine/risk_scoring/risk_scores_request_schema.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import { DataViewId } from '../../detection_engine/rule_schema';
+
+const afterKey = t.record(t.string, t.string);
+
+export type RiskScoresRequestSchema = t.TypeOf<typeof riskScoresRequestSchema>;
+export const riskScoresRequestSchema = t.exact(
+  t.partial({
+    after_keys: t.partial({
+      host: afterKey,
+      user: afterKey,
+    }),
+    data_view_id: DataViewId,
+    debug: t.boolean,
+    filter: t.unknown,
+    page_size: t.number,
+    identifier_type: t.keyof({ user: null, host: null }),
+    range: t.type({
+      start: t.string,
+      end: t.string,
+    }),
+    weights: t.array(
+      t.intersection([
+        t.partial({
+          value: t.string,
+          host: t.number,
```

Comment:
I am not sure if it possible to add here validation for number range 0..1 like (that)[https://github.com/gcanti/io-ts/issues/313#issuecomment-492784249]

It if not possible with io-ts, maybe we should add some other validation for those weight numbers?

Reply:
@nkhristinin @oatkiller I added more comprehensive schema/validation in https://github.com/elastic/kibana/pull/155966/commits/7dd726603a0bf93914afe910b121e7190a5cfa00 if you want to take another look 👍